x-pack/plugins/maps/public/components/layer_panel/flyout_footer/index.js

```diff
@@ -20,17 +26,17 @@ const mapStateToProps = state => {
 
 const mapDispatchToProps = dispatch => {
   return {
-    cancelLayerPanel: () => {
-      dispatch(updateFlyout(FLYOUT_STATE.NONE));
-      dispatch(clearTemporaryStyles());
-      dispatch(clearTemporaryLayers());
+    cancelLayerPanel: async () => {
+      await dispatch(updateFlyout(FLYOUT_STATE.NONE));
+      await dispatch(clearTemporaryLayers());
+      await dispatch(rollbackToTrackedLayerStateForSelectedLayer());
```

Comment:
Maybe we should just dispatch `rollbackToTrackedLayerStateForSelectedLayer()` from `clearTemporaryLayers()`. I can't think of a situation where we'd clear temp layers and not want to rollback tracked layer state.

Reply:
When the temporary-layer cleanup is done (https://github.com/elastic/kibana/pull/30542/files), we shouldn't call `clearTemporaryLayers` anymore at all on cancel, because there will never be any temporary-layers when the layer-panel is populated.